### DISCUSSION TITLE
Features for Event-Driven Architecture

### DISCUSSION THREAD

**Discussion by AndryHTC** [serverpod/serverpod#614] [Q&A] [answered]
Hi @vlidholt,

We are considering implementing an event-driven approach in some parts of our application. Do you believe that the "Future Calls" feature would be sufficient for handling events and reacting to them? Additionally, what is your estimate for the technical limit for a large number of concurrent events and parallel "Future Calls"?
Finally, is there a plan to add any specific features for event-driven architecture?

**Comment by vlidholt** [comment, marked as answer]
For an event-driven architecture, you probably want to use MessageCentral over future calls. It is very performant and also scales over a cluster of servers with Redis (if you need that).

There are some docs to get you started here: https://docs.serverpod.dev/concepts/streams

Also, check out the Pixorama demo. It uses streams and messaging and is very well documented: https://pixorama.live/
